ts/main.ts

```diff
@@ -0,0 +1,64 @@
+interface Anime {
+  mal_id: number;
+  title: string;
+  episodes: number;
+  images: {
+    jpg: {
+      image_url: string;
+    };
+  };
+  [key: string]: any; // To account for any additional fields
+}
+
+const animeContainer = document.querySelector(
+  '.anime-container',
+) as HTMLDivElement;
+
+async function fetchAnime(): Promise<Anime[]> {
+  try {
+    const response = await fetch('https://api.jikan.moe/v4/top/anime');
+    const data = await response.json();
+    return data.data as Anime[];
+  } catch (error) {
+    console.error('Error fetching anime data:', error);
+    return [];
+  }
```

Comment:
You shouldn't need to return anything here since the query is the data we want and if it fails, the application would stop due to having a `throw new Error` set up.